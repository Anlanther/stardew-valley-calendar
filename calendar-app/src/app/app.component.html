<nav class="navbar navbar-expand-lg navbar-dark bg-dark">
  <a class="navbar-brand">Stardew Valley Calendar</a>
  <button
    class="navbar-toggler"
    type="button"
    data-toggle="collapse"
    data-target="#navbarSupportedContent"
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
  >
    <span class="navbar-toggler-icon"></span>
  </button>

  <div
    class="collapse navbar-collapse d-flex justify-content-between"
    id="navbarSupportedContent"
  >
    <ul class="navbar-nav mr-auto">
      <button class="btn btn-link" (click)="openEditDialog()">
        @if (activeCalendar$ | async; as calendar) {
          {{ calendar.name }}
        } @else {
          Edit Calendar
        }
      </button>
    </ul>
    <div class="d-flex">
      <button class="btn btn-primary" (click)="openCreateDialog()">
        Create New
      </button>
      <button class="btn btn-danger" [disabled]="disableDelete()">
        Delete
      </button>
    </div>
  </div>
</nav>

<div class="container-xxl">
  <div class="row app-row">
    <div class="col-8 calendar-grid"><app-calendar></app-calendar></div>
    <div class="col-4"><app-day-form></app-day-form></div>
  </div>
</div>
