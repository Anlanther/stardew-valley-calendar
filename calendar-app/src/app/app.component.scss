.welcome__actions {
  gap: 2px;
}
