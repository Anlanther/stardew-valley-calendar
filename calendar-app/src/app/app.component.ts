import { Component, inject } from '@angular/core';
import { Store } from '@ngrx/store';
import { Observable } from 'rxjs';
import { AppStore } from './models/app-store.model';
import { DownloadedCalendar } from './models/downloaded-calendar.model';
import { StatusMessage } from './models/status-message.model';
import { AppActions } from './state/app.actions';
import { AppFeature } from './state/app.state';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss',
})
export class AppComponent {
  store = inject(Store<AppStore>);

  status$: Observable<StatusMessage>;

  uploadFailed = false;

  get noApi() {
    return StatusMessage.NO_API_ACCESS;
  }
  get ready() {
    return StatusMessage.READY;
  }
  get noCalendars() {
    return StatusMessage.NO_CALENDARS_AVAILABLE;
  }
  get noSelectedCalendar() {
    return StatusMessage.NO_SELECTED_CALENDAR;
  }

  constructor() {
    this.store.dispatch(AppActions.initialise());
    this.status$ = this.store.select(AppFeature.selectStatusMessage);
  }

  openCreateDialog() {
    this.store.dispatch(AppActions.createCalendar());
  }

  openSelectDialog() {
    this.store.dispatch(AppActions.selectCalendar());
  }

  onFileChange(event: any) {
    const file = event.target.files[0];
    const fileReader = new FileReader();
    fileReader.onload = () => {
      if (typeof fileReader.result === 'string') {
        const uploadedCalendar: DownloadedCalendar = JSON.parse(
          fileReader.result,
        );
        this.store.dispatch(
          AppActions.createUploadedCalendar(uploadedCalendar),
        );
      }
    };
    fileReader.readAsText(file);
  }
}
