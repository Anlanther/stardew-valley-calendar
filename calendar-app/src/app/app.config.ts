import { ApplicationConfig } from '@angular/core';
import { provideRouter } from '@angular/router';

import { provideHttpClient } from '@angular/common/http';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { provideEffects } from '@ngrx/effects';
import { provideStore } from '@ngrx/store';
import { provideStoreDevtools } from '@ngrx/store-devtools';
import { routes } from './app.routes';
import { AppEffects } from './state/app.effect';
import { AppFeature } from './state/app.state';

export const appConfig: ApplicationConfig = {
  providers: [
    provideRouter(routes),
    provideHttpClient(),
    provideAnimationsAsync(),
    provideStore({ [AppFeature.name]: AppFeature.reducer }),
    provideEffects(AppEffects),
    provideStoreDevtools({ name: 'Stardew Calendar', maxAge: 15 }),
  ],
};
