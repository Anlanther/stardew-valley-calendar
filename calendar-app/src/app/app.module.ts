import {
  provideHttpClient,
  withInterceptorsFromDi,
} from '@angular/common/http';
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { CalendarModule } from './calendar/calendar.module';
import { DayFormModule } from './day-form/day-form.module';
import { DialogsModule } from './dialogs/dialogs.module';
import { AppEffects } from './state/app.effect';
import { AppFeature } from './state/app.state';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    AppRoutingModule,
    CalendarModule,
    DayFormModule,
    DialogsModule,
    StoreModule.forRoot({ [AppFeature.name]: AppFeature.reducer }),
    EffectsModule.forRoot(AppEffects),
    StoreDevtoolsModule.instrument({
      name: 'Stardew Valley Calendar',
      maxAge: 25,
    }),
  ],
  providers: [
    provideAnimationsAsync(),
    provideHttpClient(withInterceptorsFromDi()),
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
