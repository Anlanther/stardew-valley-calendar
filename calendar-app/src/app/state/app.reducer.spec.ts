import { MOCK_CALENDARS } from '../../../tests/utils/mocks/calendars.mock';
import { MOCK_GAME_EVENTS } from '../../../tests/utils/mocks/game-events.mock';
import { Season } from '../models/season.model';
import { AppActions } from './app.actions';
import { appReducer, AppState, initialState } from './app.reducer';

describe('AppReducer', () => {
  let state: AppState;

  beforeEach(() => {
    state = initialState;
  });

  describe(AppActions.createCalendarSuccess.type, () => {
    it('should update the state', () => {
      const mockCreatedCalendar = MOCK_CALENDARS[0];
      const action = AppActions.createCalendarSuccess(mockCreatedCalendar);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateActiveCalendar.type, () => {
    it('should update the state', () => {
      const mockNewlySelectedCalendar = MOCK_CALENDARS[0];
      const action = AppActions.updateActiveCalendar(mockNewlySelectedCalendar);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.getCalendarsSuccess.type, () => {
    it('should update the state', () => {
      const mockAllAvailableCalendars = MOCK_CALENDARS;
      const action = AppActions.getCalendarsSuccess(mockAllAvailableCalendars);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateActiveDayEvents.type, () => {
    it('should update the state', () => {
      const mockGameEventsOnSelectedDate = MOCK_GAME_EVENTS;
      const action = AppActions.updateActiveDayEvents(
        mockGameEventsOnSelectedDate,
      );

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateActiveSeasonGoals.type, () => {
    it('should update the state', () => {
      state.activeCalendar = MOCK_CALENDARS[0];
      const action = AppActions.updateActiveSeasonGoals();

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateSeason.type, () => {
    it('should update the state', () => {
      const mockSelectedSeason = Season.FALL;
      const action = AppActions.updateSeason(mockSelectedSeason);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.addedEventToCalendar.type, () => {
    it('should update the state', () => {
      const mockEvent = MOCK_GAME_EVENTS[0];
      const mockCalendarToAddEventTo = MOCK_CALENDARS[0];
      const action = AppActions.addedEventToCalendar(
        mockCalendarToAddEventTo,
        mockEvent,
      );

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.addedGoalToCalendar.type, () => {
    it('should update the state', () => {
      const mockGoal = MOCK_GAME_EVENTS[0];
      const mockCalendarToAddEventTo = MOCK_CALENDARS[0];
      const action = AppActions.addedGoalToCalendar(
        mockCalendarToAddEventTo,
        mockGoal,
      );

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.deleteEventSuccess.type, () => {
    it('should update the state', () => {
      const mockDeletedGameId = MOCK_GAME_EVENTS[0].id;
      const action = AppActions.deleteEventSuccess(mockDeletedGameId);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.deleteGoalSuccess.type, () => {
    it('should update the state', () => {
      const mockDeletedGoalId = MOCK_CALENDARS[0].id;
      const action = AppActions.deleteGoalSuccess(mockDeletedGoalId);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.deleteCalendarSuccess.type, () => {
    it('should update the state', () => {
      const mockDeletedCalendarId = MOCK_CALENDARS[0].id;
      const action = AppActions.deleteCalendarSuccess(mockDeletedCalendarId);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateActiveDay.type, () => {
    it('should update the state', () => {
      const mockSelectedDay = 2;
      const action = AppActions.updateActiveDay(mockSelectedDay);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateEventSuccess.type, () => {
    it('should update the state', () => {
      const mockGameEventUpdated = MOCK_GAME_EVENTS[0];
      const action = AppActions.updateEventSuccess(mockGameEventUpdated);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateGoalSuccess.type, () => {
    it('should update the state', () => {
      const mockGameEventUpdated = MOCK_GAME_EVENTS[0];
      const action = AppActions.updateGoalSuccess(mockGameEventUpdated);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.toggleEventNav.type, () => {
    it('should update the state', () => {
      const isToggleOpen = true;
      const action = AppActions.toggleEventNav(isToggleOpen);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.toggleSeasonNav.type, () => {
    it('should update the state', () => {
      const isToggleOpen = true;
      const action = AppActions.toggleSeasonNav(isToggleOpen);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.createDefaultGameEventsSuccess.type, () => {
    it('should update the state', () => {
      const mockSystemEvents = MOCK_GAME_EVENTS;
      const action =
        AppActions.createDefaultGameEventsSuccess(mockSystemEvents);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.updateCalendarSuccess.type, () => {
    it('should update the state', () => {
      const mockCalendarWithUpdatedDetails = MOCK_CALENDARS[0];
      const mockUpdatedYear = 2;
      const action = AppActions.updateCalendarSuccess(
        mockCalendarWithUpdatedDetails,
        mockUpdatedYear,
      );

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.aPIFailed.type, () => {
    it('should update the state', () => {
      const action = AppActions.aPIFailed(true);

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
  describe(AppActions.setOfflineMode.type, () => {
    it('should update the state', () => {
      const action = AppActions.setOfflineMode();

      const actual = appReducer(state, action);

      expect(actual).toMatchSnapshot();
    });
  });
});
