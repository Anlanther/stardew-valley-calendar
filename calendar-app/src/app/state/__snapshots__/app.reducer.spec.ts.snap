// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppReducer [App] API Failed should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": true,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Added Event to Calendar should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [
    {
      "description": "This is a mock calendar for tests.",
      "filteredGameEvents": [],
      "gameEvents": [],
      "id": "1",
      "name": "Mock Calendar 1",
      "publishedAt": "",
      "systemConfig": {
        "includeBirthdays": false,
        "includeCrops": false,
        "includeFestivals": false,
      },
    },
  ],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Create Calendar Success should update the state 1`] = `
{
  "activeCalendar": {
    "description": "This is a mock calendar for tests.",
    "filteredGameEvents": [],
    "gameEvents": [],
    "id": "1",
    "name": "Mock Calendar 1",
    "publishedAt": "",
    "systemConfig": {
      "includeBirthdays": false,
      "includeCrops": false,
      "includeFestivals": false,
    },
  },
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [
    {
      "description": "This is a mock calendar for tests.",
      "filteredGameEvents": [],
      "gameEvents": [],
      "id": "1",
      "name": "Mock Calendar 1",
      "publishedAt": "",
      "systemConfig": {
        "includeBirthdays": false,
        "includeCrops": false,
        "includeFestivals": false,
      },
    },
  ],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Create Default Game Events Success should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [
    {
      "description": "Only available when raining.",
      "gameDate": {
        "day": 13,
        "id": "1",
        "isRecurring": false,
        "season": "fall",
        "year": 2,
      },
      "id": "1",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Fish for Legendary",
      "type": "user",
    },
    {
      "description": "Need 30 iron ores.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": false,
        "season": "winter",
        "year": 3,
      },
      "id": "2",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Go Mining",
      "type": "user",
    },
    {
      "description": "Buy gem.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": true,
        "season": "winter",
      },
      "id": "3",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Caroline's Birthday",
      "type": "system_birthdays",
    },
    {
      "description": "Prepare gift.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": true,
        "season": "winter",
      },
      "id": "4",
      "publishedAt": "",
      "tag": "festival",
      "title": "Feast of the Winter Star",
      "type": "system_festivals",
    },
    {
      "description": "Sell and plant more immediately.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": false,
        "season": "winter",
        "year": 4,
      },
      "id": "5",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Harvest Stawberries",
      "type": "system_crops",
    },
  ],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Delete Event Success should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Delete Event Success should update the state 2`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Get Calendars Success should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [
    {
      "description": "This is a mock calendar for tests.",
      "filteredGameEvents": [],
      "gameEvents": [],
      "id": "1",
      "name": "Mock Calendar 1",
      "publishedAt": "",
      "systemConfig": {
        "includeBirthdays": false,
        "includeCrops": false,
        "includeFestivals": false,
      },
    },
    {
      "description": "This is a second mock calendar for tests.",
      "filteredGameEvents": [],
      "gameEvents": [],
      "id": "2",
      "name": "Mock Calendar 2",
      "publishedAt": "",
      "systemConfig": {
        "includeBirthdays": false,
        "includeCrops": false,
        "includeFestivals": false,
      },
    },
  ],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Set Offline Mode should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": true,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Toggle Nav Bar should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": true,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Update Active Calendar should update the state 1`] = `
{
  "activeCalendar": {
    "description": "This is a mock calendar for tests.",
    "filteredGameEvents": [],
    "gameEvents": [],
    "id": "1",
    "name": "Mock Calendar 1",
    "publishedAt": "",
    "systemConfig": {
      "includeBirthdays": false,
      "includeCrops": false,
      "includeFestivals": false,
    },
  },
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Update Active Day should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 2,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Update Active Form Events should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": [
    {
      "description": "Only available when raining.",
      "gameDate": {
        "day": 13,
        "id": "1",
        "isRecurring": false,
        "season": "fall",
        "year": 2,
      },
      "id": "1",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Fish for Legendary",
      "type": "user",
    },
    {
      "description": "Need 30 iron ores.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": false,
        "season": "winter",
        "year": 3,
      },
      "id": "2",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Go Mining",
      "type": "user",
    },
    {
      "description": "Buy gem.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": true,
        "season": "winter",
      },
      "id": "3",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Caroline's Birthday",
      "type": "system_birthdays",
    },
    {
      "description": "Prepare gift.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": true,
        "season": "winter",
      },
      "id": "4",
      "publishedAt": "",
      "tag": "festival",
      "title": "Feast of the Winter Star",
      "type": "system_festivals",
    },
    {
      "description": "Sell and plant more immediately.",
      "gameDate": {
        "day": 1,
        "id": "2",
        "isRecurring": false,
        "season": "winter",
        "year": 4,
      },
      "id": "5",
      "publishedAt": "",
      "tag": "fishing",
      "title": "Harvest Stawberries",
      "type": "system_crops",
    },
  ],
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Update Calendar Success should update the state 1`] = `
{
  "activeCalendar": {
    "description": "This is a mock calendar for tests.",
    "filteredGameEvents": [],
    "gameEvents": [],
    "id": "1",
    "name": "Mock Calendar 1",
    "publishedAt": "",
    "systemConfig": {
      "includeBirthdays": false,
      "includeCrops": false,
      "includeFestivals": false,
    },
  },
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [
    {
      "description": "This is a mock calendar for tests.",
      "filteredGameEvents": [],
      "gameEvents": [],
      "id": "1",
      "name": "Mock Calendar 1",
      "publishedAt": "",
      "systemConfig": {
        "includeBirthdays": false,
        "includeCrops": false,
        "includeFestivals": false,
      },
    },
  ],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 2,
  },
}
`;

exports[`AppReducer [App] Update Event Success should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "spring",
    "year": 1,
  },
}
`;

exports[`AppReducer [App] Update Season should update the state 1`] = `
{
  "activeCalendar": null,
  "activeFormEvents": null,
  "apiFailed": false,
  "availableCalendars": [],
  "navBarOpen": false,
  "offlineMode": false,
  "savedSystemEvents": [],
  "selectedDate": {
    "day": 1,
    "season": "fall",
    "year": 1,
  },
}
`;
